distributed/scheduler.py

```diff
@@ -385,12 +387,81 @@ def mark_ready_to_run(self, key):
                 self.unrunnable.add(key)
             else:
                 self.stacks[new_worker].append(key)
-                self.maybe_ready.add(new_worker)
+                self.maybe_idle.add(new_worker)
                 # self.ensure_occupied(new_worker)
         else:
             self.ready.appendleft(key)
             # self.ensure_idle_ready()
 
+    def should_steal(self, key, bandwidth=100e6):
+        """ Is a key good for stealing from its chosen worker?
+
+        It must have the following attributes
+
+        1.  Not have too many dependencies
+        2.  Not be restricted to run on that worker
+        3   Take less time to transfer than to compute
+        """
+        if len(self.dependencies[key]) > 10:
+            return False
+        if key in self.restrictions and key not in self.loose_restrictions:
+            return False
+
+        nbytes = sum(self.nbytes[k] for k in self.dependencies[key])
+        transfer_time = nbytes / bandwidth
+        compute_time = self.task_duration.get(key_split(key), 1)
```

Comment:
Generally we have an estimate after we run the first task, but this does open us up to bad behavior on initial tasks.  Perhaps we just shouldn't steal until we have a guess.